python/google/protobuf/json_format.py

```diff
@@ -211,16 +213,22 @@ def _MessageToJsonObject(self, message):
     js = {}
     return self._RegularMessageToJsonObject(message, js)
 
+  def _get_field_name(self, message, field):
+    if field.containing_oneof and self.use_oneof_name:
+      name = field.containing_oneof.name
+    elif self.preserving_proto_field_name:
+      name = field.name
+    else:
+      name = field.json_name
+    return name
```

Comment:
This seems really useful, I wonder why it's not in the main repo

Reply:
yeah this is a bit wild to me that it's like this by default. it also took me a long time to find how they actually structured it.